tensorflow_probability/python/stats/sample_stats_test.py

```diff
@@ -603,6 +603,198 @@ def test_bad_axis_dynamic(self):
+    x = rng.rand(100, 10)
+    running_cov = tfp.stats.init_running_covariance((10,))
+    for sample in x:
+      running_cov = tfp.stats.update_running_covariance(running_cov, sample)
+    final_state = tfp.stats.finalize_running_covariance(running_cov, ddof=1)
+    final_cov = self.evaluate(final_state.covariance)
+    # numpy defaults to the sample covariance (ddof=1)
+    self.assertAllClose(np.cov(x.T), final_cov, rtol=1e-5)
+
+  def test_higher_rank_running_covariance(self):
+    rng = np.random.RandomState(seed=test_util.test_seed())
+    x = rng.rand(100, 5, 2)
+    running_cov = tfp.stats.init_running_covariance(tf.TensorShape([5, 2]))
+    for sample in x:
+      running_cov = tfp.stats.update_running_covariance(running_cov, sample)
+    final_state = tfp.stats.finalize_running_covariance(running_cov)
+    final_cov = self.evaluate(final_state.covariance)
+    self.assertEqual(final_cov.shape, (5, 2, 5, 2))
+
+    # reshaping to be compatible with a check against numpy
+    x_reshaped = x.reshape(100, 10)
+    final_cov_reshaped = tf.reshape(final_cov, (10, 10))
+    self.assertAllClose(np.cov(x_reshaped.T, ddof=0),
+                        final_cov_reshaped,
+                        rtol=1e-5)
+
+  def test_batched_running_covariance(self):
+    rng = np.random.RandomState(seed=test_util.test_seed())
+    x = rng.rand(100, 2, 3, 5)
+    running_cov = tfp.stats.init_running_covariance((3, 5))
+    for sample in x:
+      running_cov = tfp.stats.update_running_covariance(
+          running_cov,
+          sample,
+          axis=0)
+    final_state = tfp.stats.finalize_running_covariance(running_cov)
+    final_cov = self.evaluate(final_state.covariance)
+    self.assertEqual(final_cov.shape, (3, 5, 3, 5))
+
+    # reshaping to be compatible with a check against numpy
+    x_reshaped = x.reshape(200, 15)
+    final_cov_reshaped = tf.reshape(final_cov, (15, 15))
+    self.assertAllClose(np.cov(x_reshaped.T, ddof=0),
+                        final_cov_reshaped,
+                        rtol=1e-5)
+
+  def test_covariance_as_variance(self):
+    rng = np.random.RandomState(seed=test_util.test_seed())
+    x = rng.rand(100, 5, 2)
+    running_cov = tfp.stats.init_running_covariance(
+        tf.TensorShape([5, 2]),
+        ndims=0)
+    running_var = tfp.stats.init_running_variance(
+        tf.TensorShape([5, 2])
+    )
+    for sample in x:
+      running_cov = tfp.stats.update_running_covariance(running_cov, sample)
+      running_var = tfp.stats.update_running_variance(running_var, sample)
+    final_cov, final_var = self.evaluate([
+        tfp.stats.finalize_running_covariance(running_cov).covariance,
+        tfp.stats.finalize_running_variance(running_var).variance])
+    self.assertEqual(final_cov.shape, (5, 2))
+    self.assertAllClose(final_cov, final_var, rtol=1e-5)
+
+  def test_covariance_with_ndims(self):
+    rng = np.random.RandomState(seed=test_util.test_seed())
+    x = rng.rand(100, 3, 5, 2)
```

Comment:
Could you add an example that accumulates a final covariance shaped 3,5,2,5,2? Ie a batch of 3x 5x2 covariances? Is that supported?

Reply:
Yeah, good idea! It is supported so a test is probably warranted